pennylane/ops/qubit.py

```diff
@@ -1580,6 +1634,78 @@ def decomposition(D, wires):
         return [QubitUnitary(np.diag(D), wires=wires)]
 
 
+class QFT(Operation):
+    r"""QFT(wires)
+    Apply a quantum Fourier transform (QFT).
+
+    For the :math:`N`-qubit computational basis state :math:`|m\rangle`, the QFT performs the
+    transformation
+
+    .. math::
+
+        |m\rangle \rightarrow \frac{1}{\sqrt{2^{N}}}\sum_{n=0}^{2^{N} - 1}\omega_{N}^{mn} |n\rangle,
+
+    where :math:`\omega_{N} = e^{\frac{2 \pi i}{2^{N}}}` is the :math:`2^{N}`-th root of unity.
+
+    **Details:**
+
+    * Number of wires: Any (the operation can act on any number of wires)
+    * Number of parameters: 0
+    * Gradient recipe: None
```

Comment:
I think it'd be valuable to add a usage example here. Also, can this be used also to apply the inverse QFT? Since that is also a common use case it'd be good to have an example of that as well.

Reply:
![image](https://user-images.githubusercontent.com/49409390/106942416-2a233000-66f2-11eb-8b60-bfcf42cd61a5.png)
I added an example :slightly_smiling_face: And the inverse can be accessed using `qml.QFT().inv()`.

Reply:
Nice!!

Reply:
@trbromley can I get you to move the Example below `Args:`, just to match our other docstrings?

Reply:
Done :smile: